rtmp-stream.go

```diff
@@ -98,6 +113,78 @@ func (f *RtmpStreamFactory) startStream(conn rtmp.ClientConn, streamName string)
 	}
 }
 
+func (f *RtmpStreamFactory) ParseURLArgument(urlArg string) (string, []*url.URL, error) {
+	var host string
+	var unparsedURLs []string
+	var urls []*url.URL
+	var r = regexp.MustCompile(`{{(?P<min>[1-9][0-9]*) (?P<max>[1-9][0-9]*)}}`) // {{123 456}}
+
+	var regexPattern = "'{{(?P<min>[1-9][0-9]*) (?P<max>[1-9][0-9]*)}}'"
+	var regexInfo = fmt.Sprintf("Use regex that matches patter %v", regexPattern)
+
+	if r.MatchString(urlArg) { // URL is a template.
+		log.Infof("Processing template URL %v with regex matching. (Used regex: '%v')", urlArg, regexPattern)
+		match := r.FindStringSubmatch(urlArg)
+		min, er := strconv.Atoi(match[1])
+		if er != nil {
+			return "", nil, fmt.Errorf("Failed to parse minimal value from url %v. %v: %v", urlArg, regexInfo, er)
+		}
+		max, er := strconv.Atoi(match[2])
+		if er != nil {
+			return "", nil, fmt.Errorf("Failed to parse maximal value from url %v. %v: %v", urlArg, regexInfo, er)
+		}
+		if min > max {
+			return "", nil, fmt.Errorf("Minimal value cannot be greater than maximal value in url %v. %v.", urlArg, regexInfo)
+		}
+		if urls, er := f.generateURLs(urlArg, match[0], min, max); er == nil {
+			unparsedURLs = append(unparsedURLs, urls...)
+		} else {
+			return "", nil, fmt.Errorf("URL generation based on template URL %v failed. %v: %v", urlArg, regexInfo, er)
+		}
+	} else { // No matching regex expression found in url argument. URL is not a template. Returning it as it is.
+		log.Infof("URL is not a template. Parsing URL %v without regex matching. (Used regex: '%v')", urlArg, regexPattern)
+		unparsedURLs = append(unparsedURLs, urlArg)
+	}
+
+	var errorMsg string
+	for _, unparsedURL := range unparsedURLs {
+		parsedURL, err := url.Parse(unparsedURL)
+		if err != nil {
+			if len(errorMsg) == 0 {
```

Comment:
For the issue of collecting multiple errors, there is the type golib.MultiError. Declare value and use Add() method, afterwards use NilOrError() to get a combined error (or nil, if there is no error).
I suggest returning the resulting error immediately after the loop, if it is non-nil.